langchain4j-weaviate/src/main/java/dev/langchain4j/store/embedding/WeaviateEmbeddingStoreImpl.java

```diff
@@ -0,0 +1,192 @@
+      .stream()
+      .findFirst()
+      .orElse(null);
+  }
+
+  @Override
+  public List<String> addAll(List<Embedding> embeddings) {
+    return addAll(embeddings, null);
+  }
+
+  @Override
+  public List<String> addAll(List<Embedding> embeddings, List<TextSegment> embedded) {
+    return addAll(null, embeddings, embedded);
+  }
+
+  @Override
+  public List<EmbeddingMatch<TextSegment>> findRelevant(Embedding referenceEmbedding, int maxResults) {
+    return findRelevant(referenceEmbedding, maxResults, DEFAULT_MIN_CERTAINTY);
+  }
+
+  @Override
+  public List<EmbeddingMatch<TextSegment>> findRelevant(
+    Embedding referenceEmbedding,
+    int maxResults,
+    double minCertainty
+  ) {
+    val result = client
+      .graphQL()
+      .get()
+      .withClassName(objectClass)
+      .withFields(
+        Field.builder().name(METADATA_TEXT_SEGMENT).build(),
+        Field
+          .builder()
+          .name(ADDITIONALS)
+          .fields(
+            Field.builder().name("id").build(),
+            Field.builder().name("certainty").build(),
+            Field.builder().name("vector").build()
+          )
+          .build()
+      )
+      .withNearVector(
+        NearVectorArgument
+          .builder()
+          .vector(referenceEmbedding.vectorAsList().toArray(new Float[0]))
+          .certainty((float) minCertainty)
+          .build()
+      )
+      .withLimit(maxResults)
+      .run();
+
+    if (result.hasErrors()) {
+      return emptyList();
+    }
+
+    val resGetPart = ((Map<String, Map>) result.getResult().getData()).entrySet().stream().findFirst();
+    if (!resGetPart.isPresent()) {
+      return emptyList();
+    }
+
+    val resItemsPart = resGetPart.get().getValue().entrySet().stream().findFirst();
+    if (!resItemsPart.isPresent()) {
+      return emptyList();
+    }
+
+    val resItems = ((Map.Entry<String, List<Map<String, ?>>>) resItemsPart.get()).getValue();
+
+    return resItems.stream().map(WeaviateEmbeddingStoreImpl::toEmbeddingMatch).collect(Collectors.toList());
+  }
+
+  private List<String> addAll(List<String> ids, List<Embedding> embeddings, List<TextSegment> embedded) {
+    if (embedded != null && embeddings.size() != embedded.size()) {
+      throw new IllegalArgumentException("The list of embeddings and embedded must have the same size");
+    }
+
+    val resIds = new ArrayList<String>();
+    val objects = new ArrayList<WeaviateObject>();
+    for (int i = 0; i < embeddings.size(); i++) {
+      val id = ids != null ? ids.get(i) : embedded != null ? generateUUI(embedded.get(i).text()) : generateRandomId();
+      resIds.add(id);
+      objects.add(buildObject(id, embeddings.get(i), embedded != null ? embedded.get(i).text() : null));
+    }
+
+    client
+      .batch()
+      .objectsBatcher()
+      .withObjects(objects.toArray(new WeaviateObject[0]))
+      .withConsistencyLevel(ConsistencyLevel.ALL)
```

Comment:
Should we move this to the constructor parameter?

Reply:
what is consistency level? 😄

Reply:
As far as I understand, this enum specifies how many replicas should persist and acknowledge before run() method returns. Either all of them, one or a quorum. It would be nice to give this control to the user of the library to decide.

Reply:
done